### PR TITLE
Merge google -> main

### PR DESCRIPTION
**Pull request by ThomasRaoux**
* c476f2881 Fix bad formatting added during merge
* 26603e37e Synchronize submodules with LLVM at llvm/llvm-project@b2780cd744ea
* b8e8d24ae Integrate LLVM at llvm/llvm-project@b2780cd744ea
* c2ee40146 Integrate LLVM at llvm/llvm-project@4750a8b1bcde
* 4577036e8 Integrate LLVM at llvm/llvm-project@a76df78470d7
* 862bd71f4 Merge pull request #5683 from ThomasRaoux:main-to-google

### CODE REVIEW

File: iree/base/logging.cc [hunk 1]
```diff
@@ -108,7 +108,7 @@ void LogMessage::EmitLogMessage() {
   {
     // NOTE: this truncates. That's fine for now and stderr is still usable.
     char str_buffer[512];
-    int str_length = snprintf(str_buffer, sizeof(str_buffer), "%s:%d] %s\n",
+    snprintf(str_buffer, sizeof(str_buffer), "%s:%d] %s\n",
                               file_name_, line_, str().c_str());
```
Comment (ScottTodd) [inline iree/base/logging.cc:112]: run `clang-format` (since the unused variable was dropped)

Comment (ThomasRaoux) [reply iree/base/logging.cc:112]: I sent a cl. Should i just add a commit here instead?

Comment (ScottTodd) [reply iree/base/logging.cc:112]: Either is fine.

File: iree/compiler/Conversion/LinalgToLLVM/LinalgVectorizePass.cpp [hunk 2]
```diff
@@ -82,6 +82,12 @@ void LinalgVectorizationPass::runOnFunction() {
       op->replaceAllUsesWith(contract);
   });
 
+  // Lowering transfer ops before unrolling as unrolling doesn't support
+  // transpose.
+  OwningRewritePatternList transferOpLowering(&getContext());
+  vector::populateVectorTransferLoweringPatterns(transferOpLowering);
+  (void)applyPatternsAndFoldGreedily(funcOp, std::move(transferOpLowering));
```
Comment (ScottTodd) [inline iree/compiler/Conversion/LinalgToLLVM/LinalgVectorizePass.cpp:89]: I think @GMNGeoffrey wanted us to follow-up on these casts to `void`, but they shouldn't stop this merge.

**Review by ScottTodd — APPROVED** [review]

File: iree/compiler/Conversion/LinalgToLLVM/LinalgVectorizePass.cpp (hunk 2 above)
Comment (ThomasRaoux) [reply iree/compiler/Conversion/LinalgToLLVM/LinalgVectorizePass.cpp:89]: Sure, @GMNGeoffrey let me know what you think. I can fix it although it is odd to just fix this spot. Also I'm hoping to remove this piece of code in the near future as it was a workaround to unblock integration.

**Pull request merged**